_posts/2019-05-10-continuous-deployment-the-first-step-on-the-road-to-recovery.markdown

```diff
@@ -0,0 +1,63 @@
+# Continuous Deployment - the first step on your road to recovery
+
+** HMM, does this still fit?**
+[Systems thinking]() in DevOps is about improving the flow of work into production.  Hand-offs, re-work and batching are all enemies of flow.  Teams don't intentionally look to reduce flow.  Hand-offs and approvals are mechanisms introduced in an attempt to tackle the risks of delivery.  Value is generated through change.  Change is the enemy of stability.  XXX said "Systems architecture is a reflection of organisational structure".  ***I say the deployment pipeline is a reflection of the reporting structure.***.  Continuous deployment is a goal for most teams; a lofty ambition that other people achieve.  For Mechanical Rock, continuous deployment is our starting point.  And continuous deployment starts with continuous delivery and a suitable branching strategy.
+
+Your branching strategy is a fundamental decision for development teams and has profound effects upon your working practices.  Making a poor choice has knock on effects that can limit improvements in other areas.  
+
+The goal of [continuous delivery](https://continuousdelivery.com/) is:
+> To make deployments—whether of a large-scale distributed system, a complex production environment, an embedded system, or an app—predictable, routine affairs that can be performed on demand.
+> We achieve all this by ensuring our code is always in a deployable state
+
+Or more simply: “if it hurts, do it more often, and bring the pain forward.”
+
+# Damming the Flow
+
+A common branching strategy adopted by many teams is git flow.  
+
+[!GitFlow diagram](../img/continuous-deployment-the-first-step)*
+
+If you aren't familiar with git flow, [Atlassian have an excellent introduction](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
+
+The goals for GitFlow are in direct conflict with the goals of Continuous Delivery and maximising flow, as [highlighted by the originator](https://nvie.com/posts/a-successful-git-branching-model/):
```

Comment:
You might want to illustrate the difference between Continuous Integration/Deployment/Delivery, because the main question that is buzzing in my mind here is what all the subtle differences are.

My feeling is you could follow a structure where-by;
- mention that you have had a lot of discussions / observations of different branching models and there effect on development
- you introduce the idea of flow, and why you feel it is important to increasing velocity
- mention continous delivery, and why you feel it has a larger impact on velocity, compare to say CI and the other CD's.
- mention what branching pattern's you feel inhibit/enhance flow.

Reply:
:thinking:

Reply:
I'm assuming a level of understanding, and there are a lot of references.  If people have questions, they can go off an do their own research first.